spark/unitycatalog/src/test/java/com/sparkuctest/UnityCatalogSupport.java

```diff
@@ -0,0 +1,266 @@
+      return socket.getLocalPort();
+    }
+  }
+
+  /**
+   * Starts the Unity Catalog server before all tests.
+   * IMPORTANT: Starts the server BEFORE calling other setup to ensure
+   * the server is running when SharedSparkSession creates the SparkSession.
+   */
+  @Before
+  public void setupUnityCatalog() throws Exception {
+    // Create temporary directory for UC server data
+    ucTempDir = Files.createTempDirectory("unity-catalog-test-").toFile();
+    ucTempDir.deleteOnExit();
+
+    // Find an available port
+    ucPort = findAvailablePort();
+
+    // Set up server properties
+    Properties serverProps = new Properties();
+    serverProps.setProperty("server.env", "test");
+
+    // Start UC server with configuration
+    ServerProperties initServerProperties = new ServerProperties(serverProps);
+
+    UnityCatalogServer server = UnityCatalogServer.builder()
+        .port(ucPort)
+        .serverProperties(initServerProperties)
+        .build();
+
+    server.start();
+    ucServer = server;
+
+    // Poll for server readiness by checking if we can create an API client and query catalogs
+    int maxRetries = 30;
+    int retryDelayMs = 500;
+    boolean serverReady = false;
+    int retries = 0;
+
+    while (!serverReady && retries < maxRetries) {
+      try {
+        ApiClient testClient = new ApiClient();
+        testClient.setScheme("http");
+        testClient.setHost("localhost");
+        testClient.setPort(ucPort);
+        CatalogsApi catalogsApi = new CatalogsApi(testClient);
+        catalogsApi.listCatalogs(null, null); // This will throw if server is not ready
+        serverReady = true;
+      } catch (Exception e) {
+        Thread.sleep(retryDelayMs);
+        retries++;
+      }
+    }
+
+    if (!serverReady) {
+      throw new RuntimeException(
+          "Unity Catalog server did not become ready after " + (maxRetries * retryDelayMs) + "ms");
+    }
+
+    // Create the catalog and default schema in the UC server
+    ApiClient client = new ApiClient();
+    client.setScheme("http");
+    client.setHost("localhost");
+    client.setPort(ucPort);
+
+    CatalogsApi catalogsApi = new CatalogsApi(client);
+    SchemasApi schemasApi = new SchemasApi(client);
+
+    // Create catalog
+    catalogsApi.createCatalog(
+        new CreateCatalog()
+            .name(getUnityCatalogName())
+            .comment("Test catalog for Delta Lake integration")
+    );
+
+    // Create default schema
+    schemasApi.createSchema(
+        new CreateSchema()
+            .name("default")
+            .catalogName(getUnityCatalogName())
+    );
+
+    logger.info("Unity Catalog server started and ready at " + getUnityCatalogUri());
+    logger.info("Created catalog '" + getUnityCatalogName() + "' with schema 'default'");
+  }
+
+  /**
+   * Stops the Unity Catalog server after all tests.
+   */
+  @After
+  public void tearDownUnityCatalog() throws Exception {
+    if (ucServer != null) {
+      ucServer.stop();
```

Comment:
Do we need to make it stop quitely ?  Since there are following resources that we need to close/clean.